core-library/src/main/java/org/silverpeas/core/admin/service/GroupSynchronizationRule.java

```diff
@@ -365,6 +372,31 @@ private static String getRuleExpression(
     return userIds;
   }
 
+  private List<String> getUserIdsByExtraFormFieldValue(String fieldName, String fieldValue) {
+    List<String> userIds = new ArrayList<>();
+    PublicationTemplate directoryTemplate =
+        PublicationTemplateManager.getInstance().getDirectoryTemplate();
+    if (directoryTemplate != null) {
+      try {
+        List<DataRecord> records = directoryTemplate.getRecordSet().getRecords(fieldName, fieldValue);
+        for (DataRecord record : records) {
+          String userId = record.getId();
+          if (isSharedDomain) {
+            userIds.add(userId);
+          } else {
+            User user = User.getById(userId);
+            if (user != null && user.getDomainId().equals(group.getDomainId())) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Refactor this code to not nest more than 4 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS134)